optimum/exporters/onnx/base.py

```diff
@@ -344,16 +344,26 @@ class OnnxConfigWithPast(OnnxConfig, ABC):
     Inherits from [`~exporters.onnx.OnnxConfig`]. A base class to handle the ONNX configuration of decoder-only models.
     """
 
-    PAD_ATTENTION_MASK_TO_MATCH_TOTAL_SEQUENCE_LENGTH = True
+    PAD_ATTENTION_MASK_TO_MATCH_TOTAL_SEQUENCE_LENGTH: bool = True
+    USE_PAST_IN_INPUTS: Optional[bool] = None
+    USE_PRESENT_IN_OUTPUTS: Optional[bool] = None
 
     def __init__(
         self,
         config: "PretrainedConfig",
         task: str = "default",
         patching_specs: List[PatchingSpec] = None,
         use_past: bool = False,
+        use_past_in_inputs: Optional[bool] = None,
+        use_present_in_outputs: Optional[bool] = None,
     ):
         self.use_past = use_past
+        if use_past_in_inputs is None:
+            use_past_in_inputs = self.USE_PAST_IN_INPUTS
+        if use_present_in_outputs is None:
+            use_present_in_outputs = self.USE_PRESENT_IN_OUTPUTS
+        self.use_past_in_inputs = use_past if use_past_in_inputs is None else use_past_in_inputs
+        self.use_present_in_outputs = use_past if use_present_in_outputs is None else use_present_in_outputs
```

Comment:
Would it be worth it to log a warning when `use_past` is equal to the complementary of `use_past_in_inputs`/`use_present_in_outputs`?

Reply:
Done!